### ISSUE TITLE
Break`basics` into it's own type+decoder?

### ISSUE THREAD

**Issue by mikeonslow** [jpierson/x-elm-resume#5]
https://github.com/jpierson/x-elm-resume/blob/1c679d5d0af48486bbe07c0717aab85387bf32e6/ElmResume/JsonResumeDecoder.elm#L56-L60

Looking at this, you _may_ want to just break `basics` out into it's own decoder with 5 optional fields, you could do that like this:
```
resumeBasicsDecoder : Decoder ResumeData.Basics
resumeBasicsDecoder = 
  decode ResumeData.Basics 
    |> optional "name" justString Nothing
    |> optional "email" justString Nothing
    |> optional "phone" justString Nothing
    |> optional "label" justString Nothing
    |> optional "summary" justString Nothing
```
You would also need to make a type for `Basics` in the `ElmResume.ResumeData` module to make this work but some benefits would be a bit of gained modularity (`Basics` could be broken out like `WorkData` etc.) and the ability to also make the `basics` field optional in totality (not sure if that's a good idea though).

**Comment by jpierson** [comment]
Overall it seems this may be a good pattern in general. Make a separate type that represents the actual structure of the Json object even if it's a subset and then transform it later if needed.

**Comment by mikeonslow** [comment]
Yup, that's totally the idea. Essentially I break each piece of data domain into it's own module with helpers, decoders, encoders etc. and then use them with one another. It's a lot to explain in a post but I can show you Thursday if you are interested.
